### PR TITLE
added react and webpack, moved rendering of main menu from html to tsx

### PR DESCRIPTION
**Pull request by algiokas**

### CODE REVIEW

**Comment by KaiSforza** [comment]
rebasing this, alex.

**Comment by KaiSforza** [comment]
i think this will fail, though.

File: index.html [hunk 1]
```diff
@@ -4,26 +4,10 @@
     <link href="https://fonts.googleapis.com/css2?family=Stalinist+One&display=swap" rel="stylesheet"> 
     <title>Durak Project</title>
   </head>
-  <body>
+  <body>
```
Comment (KaiSforza) [inline index.html:7]: wtf fam stop the trailing spaces

File: package.json [hunk 2]
```diff
@@ -4,14 +4,14 @@
   "description": "Durak card game using electron",
   "main": "dist/main.js",
   "scripts": {
-    "build": "tsc --build",
+    "build": "webpack --config ./webpack.config.js",
```
Comment (KaiSforza) [inline package.json:7]: need to add `webpack.config.js`

File: package.json [hunk 3]
```diff
@@ -4,14 +4,14 @@
...
     "build:watch": "tsc --watch",
     "clean": "tsc --build --clean",
     "rebuild": "npm run clean && npm run build",
     "test": "jest --coverage",
     "watch": "jest --watch",
     "prestart": "npm run build",
-    "start": "electron ."
+    "start": "npm run build && electron ./dist/main.js"
```
Comment (KaiSforza) [inline package.json:14]: remove `npm run build &&`, we are already doing that in `prestart`

**Review by KaiSforza — CHANGES_REQUESTED** [review]

File: index.html (hunk 1 above)
Comment (algiokas) [reply index.html:7]: don't worry about this, I'm going to set up auto-generation for this index file anyway so it's pretty much irrelevant

**Pull request merged**